api/v1beta1/common.go

```diff
@@ -16,48 +16,104 @@ limitations under the License.
 
 package v1beta1
 
-type CommonParameters struct {
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type ExtraConfigmapsMounts struct {
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength:=253
+	// The name of an existing config map for mounting.
+	Name string `json:"name"`
+
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:validation:Required
+	// Path within the container at which the volume should be mounted.
+	MountPath string `json:"mountPath"`
+
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
 	// +kubebuilder:validation:optional
-	// +kubebuilder:default=false
-	// +optional
-	// Use with caution! This parameter specifies whether test-operator should spawn test
-	// pods with allowedPrivilegedEscalation: true and the default capabilities on
-	// top of capabilities that are usually needed by the test pods (NET_ADMIN, NET_RAW).
-	// This parameter is deemed insecure but it is needed for certain test-operator
-	// functionalities to work properly (e.g.: extraRPMs in Tempest CR, or certain set
-	// of tobiko tests).
-	Privileged bool `json:"privileged"`
+	// +kubebuilder:default:=""
+	// Config map subpath for mounting, defaults to configmap root.
+	SubPath string `json:"subPath"`
 }
 
-type WorkflowCommonParameters struct {
```

Comment:
Yes, I know. The thing is that we need to have the `privileged:` parameter with type `*bool` in the `workflow:` section. The reason for this is that we need to be able to distinguish whether the user set `spec.workflow.privileged` or not.

The logic that decides whether to use to value from the workflow section is build on the idea of workflow parameters being pointers [1]. We have a PR in progress that will unify the whole processing of the `workflow` section:

- https://github.com/openstack-k8s-operators/test-operator/pull/168

Because of that I suggest let's add the `WorkflowCommonParameters` section and deal with the rest in the PR 168. This is PR is more than helpful and it would be great if we could merge it.

[1] https://github.com/openstack-k8s-operators/test-operator/blob/main/controllers/tobiko_controller.go#L310

Reply:
ack I changed it for all workflow parameters.